packages/block-editor/src/components/writing-flow/index.js

```diff
@@ -72,6 +72,28 @@ export function isNavigationCandidate( element, keyCode, hasModifier ) {
 	return tagName !== 'INPUT' && tagName !== 'TEXTAREA';
 }
 
+/**
+ * From the point determined by the x and y position, search using the increment until finding
+ * a tabbable text field or until the number of searches expires.
+ *
+ * @param {number} xPosition The horizontal starting position.
+ * @param {number} yPosition The vertical starting position.
+ * @param {number} increment The size of the increment from the starting position.
+ * @param {number} searches  The number of searches to perform.
+ */
+function searchVerticallyForTabbableTextField( xPosition, yPosition, increment = 5, searches = 10 ) {
+	if ( searches === 0 ) {
+		return;
+	}
+
+	const element = document.elementFromPoint( xPosition, yPosition );
+	if ( element && isTabbableTextField( element ) ) {
```

Comment:
This works ok for now, but would fail if the element at the point was some formatting. A solution would be to check parent elements as well.

There is also `elementsFromPoint`, which although marked on MDN as experimental, seems to be pretty widely supported:
https://developer.mozilla.org/en-US/docs/Web/API/DocumentOrShadowRoot/elementsFromPoint